pkg/config/context.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+)
+
+// SyncContextsAndServers populate or sync contexts and servers
+func SyncContextsAndServers() error {
+	cfg, err := config.GetClientConfig()
+	if err != nil {
+		return errors.Wrap(err, "failed to get client config")
+	}
+
+	config.PopulateContexts(cfg)
+
+	for _, c := range cfg.KnownContexts {
```

Comment:
@marc pkg/config/In init.go we were already doing this sync of contexts and servers
 // contexts could be lost when older plugins edit the config, so populate them from servers
	addedContexts := config.PopulateContexts(c)
	
I removed the use of StoreClientConfig and used the Config APIs directly.

Reply:
Ok, now I understand two things:
1. `StoreClientConfig()` would of course store the config, which `config.PopulateContexts(c)` does not do, so in this PR you save each context in the config file by looping and calling `config.SetContext()`
1. `StoreClientConfig()` also made sure any missing servers were created based on the contexts, and now the loop achieves the same thing by the same calls to `config.SetContext()`

This is great, but could you add a comment before the loop to say something like `// Now write the context to the configuration file.  This will also create any missing server for its corresponding context`

Also, I think there still one aspect missing:
- `config.PopulateContexts(c)` might have change the value of the  currentContext; this is not persisted in the config file because the loop never sets the currentContext

I was able to reproduce that last point as follows:
1. manually delete a context from `~/.config/tanzu/config-ng.yaml`
2. set the deleted context as the `current` in `~/.config/tanzu/config.yaml`
3. => So now we have a server that is active that does not exist as a context.
4. Now run `tz context list` (or any other command).
5. => before this PR, the missing context would be added and be set to active, but with this PR the missing context is added but is not set to active.

Reply:
Thanks for the detailed review.  I have added comments for SetContext and updated the code to use SetActiveContext.